internal/status.go

```diff
@@ -0,0 +1,118 @@
+package internal
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/dustin/go-humanize"
+)
+
+type StatusLine struct {
+	resources []Resource
+	indexCh   chan int
+}
+
+var spinChars = [5]string{"-", "\\", "|", "/", "-"}
+
+func (st *StatusLine) run() {
+	startTime := time.Now()
+
+	resourceSizes, totalBytes, sizingSuccess := getResourcesSizes(st, 10000)
+
+	var bytesDownloaded int64
+	resourcesDownloaded := 0
+	go func() {
+		spinI := 0
+
+		for {
+			var i int
+			select {
+			case i = <-st.indexCh:
+			case <-time.After(50 * time.Millisecond):
+				i = -1
+			}
+			if i != -1 {
+				bytesDownloaded += resourceSizes[i]
+				resourcesDownloaded++
+			}
+			anyRemaining := resourcesDownloaded < len(st.resources)
+
+			spinI += 1
+			if spinI == len(spinChars) {
+				spinI = 0
+			}
+
+			line := composeStatusString(bytesDownloaded, totalBytes, resourcesDownloaded, len(st.resources), sizingSuccess, spinChars[:], spinI, startTime, anyRemaining, 20)
+			fmt.Print(line)
+			if !anyRemaining {
+				fmt.Println()
+				break
+			}
+
+		}
+	}()
+
+}
+
+func getResourcesSizes(st *StatusLine, timeoutMs int) ([]int64, int64, bool) {
```

Comment:
This should be a method on the struct like `run`, this way the caller just have to call object methods.